contrib/database/sql/sql.go

```diff
@@ -12,23 +12,34 @@
 package sql
 
 import (
+	"context"
 	"database/sql"
 	"database/sql/driver"
 	"errors"
+	"reflect"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql/internal"
 )
 
+var driverTypeToName = make(map[reflect.Type]string)
```

Comment:
This is a bit confusing and takes a while to wrap your head around. Can we simplify it somehow? Maybe a structure would be clearer?
```go
// registeredDrivers holds a registry of all drivers registered via the sqltrace package.
var registeredDrivers struct {
    // keys maps driver types to their registered names.
    keys map[reflect.Type]string
    // drivers maps keys to their registered driver.
    drivers map[string]driver.Driver
    // configs maps keys to their registered configuration.
    configs map[string]*registerConfig
}
```
Alternatively, we could include `driver.Driver` into the `registerConfig` and remove the `drivers` field from above completely.

You could also add methods to this structure to help you with registration and lookup. It would contain the logic a bit better and make the rest of the code easier to read.

Reply:
Tried it this way and it's much clearer IMO. A very helpful suggestion, thanks!